src/pal/src/misc/cgroup.cpp

```diff
@@ -315,6 +468,36 @@ class CGroup
         return cgroup_path;
     }
 
+    static char* SearchForFile(char* search_root, const char* possible_files[], size_t possible_files_len)
+    {
+        if (search_root == nullptr)
+        {
+            return nullptr;
+        }
+
+        for (size_t i = 0; i < possible_files_len; i++)
+        {
+            const char* possible_file = possible_files[i];
+            size_t len = strlen(search_root);
+            len += strlen(possible_file);
+            char* full_filename = (char*)PAL_malloc(len+1);
+            if (full_filename == nullptr)
+            {
+                return nullptr;
+            }
+            strcpy_s(full_filename, len+1, search_root);
+            strcat_s(full_filename, len+1, possible_file);
```

Comment:
Might be better to use `snprintf()` on a buffer declared with size `PATH_MAX`, and `strdup()` when found.  Saves a roundtrip to malloc.